plugins/rating/services/RatingService.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * Allows user to manipulate their entry rating
+ *
+ * @service rating
+ * @package plugins.rating
+ * @subpackage api.services
+ */
+
+class RatingService extends KalturaBaseService
+{
+	public function initService($serviceId, $serviceName, $actionName)
+	{
+		parent::initService($serviceId, $serviceName, $actionName);
+		
+		if(!RatingPlugin::isAllowedPartner($this->getPartnerId()))
+		{
+			throw new KalturaAPIException(KalturaErrors::FEATURE_FORBIDDEN, RatingPlugin::PLUGIN_NAME);
+		}
+		
+		if ((!kCurrentContext::$ks_uid || kCurrentContext::$ks_uid == "") && $actionName != "getRatingCounts")
+		{
+			throw new KalturaAPIException(KalturaErrors::INVALID_USER_ID);
```

Comment:
This exception does not imply on the real cause. 
Must add the following info to the API Exception - "Anonymous user is not allowed"